etna/metrics/functional_metrics.py

```diff
@@ -1,14 +1,33 @@
+from enum import Enum
 from functools import partial
-from typing import List
+from typing import Sequence
 from typing import Union
 
 import numpy as np
 from sklearn.metrics import mean_squared_error as mse
+from typing_extensions import assert_never
 
-ArrayLike = List[Union[float, List[float]]]
+ArrayLike = Union[float, Sequence[float], Sequence[Sequence[float]]]
```

Comment:
Is "float" really "ArrayLike"?

Reply:
I tried to follow smth like this: https://stackoverflow.com/a/40380014